saw-core-coq/src/Verifier/SAW/Translation/Coq/Term.hs

```diff
@@ -479,44 +478,44 @@ translatePiBinder ::
   TermTranslationMonad m => (LocalName, Term) -> m [Coq.PiBinder]
 translatePiBinder (n, ty) =
   translateBinder n ty >>= \case
-    Left (n',ty')
+    (n',ty',[])
       | n == "_"  -> return [Coq.PiBinder Nothing ty']
       | otherwise -> return [Coq.PiBinder (Just n') ty']
-    Right (n',ty',nh,nhty) ->
-      return [Coq.PiBinder (Just n') ty', Coq.PiImplicitBinder (Just nh) nhty]
+    (n',ty',nhs) ->
+      return $ Coq.PiBinder (Just n') ty' :
+               map (\(nh,nhty) -> Coq.PiImplicitBinder (Just nh) nhty) nhs
 
 translateBinder ::
```

Comment:
Please add haddock to at least the functions you have modified, to explain what those functions do and what the inputs and outputs represent. E.g., after reading through `translateBinder`, I eventually figured out that the returned list of identifiers plus types as the typeclass assumptions, but that should be in the haddock.

Reply:
Done! In 05ef2cbeacca2d7b55f66ae87ecf890c23455b52 I added Haddock for all binder-related definitions in the Coq translator (most of which I modified).